src/tutorials/0004-mutable-file-system/05.md

```diff
@@ -23,20 +23,36 @@ such as `/catPics`,
 `files.ls` produces an array of objects, one for each file or directory
 contained in the directory you're inspecting, with the following properties:
 
-- `name` (default): the file's name
-- `type` (optional with `{ long: true }`): the object's type (`0` - file or `1` - directory)
-- `size` (optional with `{ long: true }`): the size of the file in bytes
-- `hash` (optional with `{ long: true }`): the cryptographic hash, or Content Identifier (CID), that represents your file in IPFS
-
-*Gotcha!* To return values for *all* of those properties, you must use the
-option `{ long: true }`. Otherwise all properties except for the `name` field will be
-present but their values will be returned as `0` (for `size` and `type`) or `""` (for `hash`).
+- `name`: the file's name
+- `type`: the object's type (`0` - file or `1` - directory)
+- `size`: the size of the file in bytes
+- `cid`: the Content Identifier (CID), that identifies your file in IPFS
 
 If we wanted to view all available details for the contents of a `/catPics`
 directory, we could do it like this:
 
 ```js
-await ipfs.files.ls('/catPics', { long: true })
+ipfs.files.ls('/catPics')
+```
+
+Because the `files.ls` method returns an [`Async Iterable`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for-await...of), you'll need to call the iterable multiple times until you have all the results.
+
+For this, we can use a `for await...of` loop:
+
+```javascript
+const result = []
+
+for await (const resultPart of ipfs.files.ls('/catPics')) {
+    result.push(resultPart)
+}
+```
+
+To make things easier, we can use the `it-all` package that does this automatically:
+
+```javascript
+const all = require('it-all')
```

Comment:
If not already explained, this `require` magic in the browser needs some love.

Reply:
We will change the variable to be global just like `ipfs` and leave a note about this to the user.

Reply:
ooh, that's a good idea!